renovate/action.yml

```diff
@@ -0,0 +1,138 @@
+name: 'Renovate with Dynamic Secrets'
+description: 'Run Renovate with auto-detected RENOVATE_* secrets from GitHub Secrets'
+author: 'EIDP'
+
+branding:
+  icon: 'refresh-cw'
+  color: 'blue'
+
+inputs:
+  github-app-id:
+    description: 'The ID of the GitHub App used for Renovate'
+    required: true
+  
+  github-app-pem:
+    description: 'The private key (in PEM format) of the GitHub App used for Renovate'
+    required: true
+  
+  all-secrets:
+    description: >-
+      JSON string of all secrets (use toJSON(secrets)). You can also manually specify
+      a JSON object. Just make sure the keys start with "RENOVATE_", e.g. (remove the space between
+      the curly brackets if copying):)
+
+      {
+        "RENOVATE_DOCKER_USERNAME": "${ { secrets.RENOVATE_DOCKER_USERNAME } }",
+        "RENOVATE_DOCKER_PASSWORD": "${ { secrets.RENOVATE_DOCKER_PASSWORD } }",
+        "RENOVATE_NPM_TOKEN": "${ { secrets.RENOVATE_NPM_TOKEN } }"
+      }
+    required: true
+  
+  renovate-config-file-path:
+    description: 'The full path to the Renovate configuration file, relative to the root'
+    required: false
+    default: 'renovate.json'
+  
+  renovate-image:
+    description: |
+      The Renovate Docker image name to use.
+      If omitted, uses ghcr.io/renovatebot/renovate.
+    required: false
+    default: 'ghcr.io/renovate/renovate'
+  
+  renovate-version:
+    description: |
+      The Renovate version to use.
+      If omitted, uses the default version Docker tag.
+    required: false
+    default: '41'
```

Comment:
Nit: this value is going to go out of date real soon (Renovate releases new versions very often). We could opt not to make the Renovate version configurable in order to keep this maintainable? Either that or this value needs to somehow be maintained by Renovate as well.

Reply:
Agree, renovate deems it to be best practice to pass along a specific version. I can leave it empty as default for now.